### PR TITLE
Abstract file storage

### PR DESCRIPTION
**Pull request by johnholdun**
This commit offers in-memory and filesystem-based storage for Post content,
and activates a different strategy based on the current environment.
This commit also tests those strategies and deletes old, unused scaffolding.

### CODE REVIEW

File: spec/models/amazon_file_store.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require_relative '../../app/models/amazon_file_store'
```
Comment (johnholdun) [inline spec/models/amazon_file_store.rb]: I set up these test files so that they can be run without loading Rails

Comment (johnholdun) [inline spec/models/amazon_file_store.rb]: For MAXIMUM SPEED

File: .rspec [hunk 2]
```diff
@@ -1,2 +1 @@
 --color
---require spec_helper
```
Comment (johnholdun) [inline .rspec]: Don't necessarily require spec_helper FOR MAXIMUM SPEED

File: .gitignore [hunk 3]
```diff
@@ -2,4 +2,5 @@
 tmp
 *.db
 db/*.sqlite3
-log/*.log
\ No newline at end of file
+log/*.log
+public/user_files
```
Comment (johnholdun) [inline .gitignore]: This is where files are saved in dev mode by default

File: spec/features/user_spec.rb [hunk 4]
```diff
@@ -1,13 +1,17 @@
+require 'spec_helper'
+
 RSpec.describe 'user actions', type: :feature do
+  let(:username) { 'jeeplanger' }
```
Comment (johnholdun) [inline spec/features/user_spec.rb]: This is a work in progress :joy:

**Comment by negatendo** [comment]
Cool! The straight S3 approach made starting things easy, but I can see how
it was far from the fastest way. This looks great!

**Comment by johnholdun** [comment]
😹 hooray! This will also make testing much easier, as we don't have to continually write and delete objects to a different S3 bucket (or, in my last few tries, the prod S3 bucket)

**Pull request merged**